File: benchmark/output/python-pygame/gemini/bouncing_ball.py
import pygame
import sys

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Bouncing Ball")
clock = pygame.time.Clock()

# Ball state
x, y = 400.0, 300.0
radius = 16
vx, vy = 150.0, 150.0

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    # Update position
    x += vx * dt
    y += vy * dt

    # Collision detection with edges
    if x - radius < 0:
        x = radius
        vx *= -1
    elif x + radius > 800:
        x = 800 - radius
        vx *= -1

    if y - radius < 0:
        y = radius
        vy *= -1
    elif y + radius > 600:
        y = 600 - radius
        vy *= -1

    # Draw
    screen.fill((0, 0, 0))
    pygame.draw.circle(screen, (255, 255, 255), (int(x), int(y)), radius)
    pygame.display.flip()

pygame.quit()
sys.exit()

File: benchmark/output/lua-love/openai/bouncing_ball.lua
local x = 400
local y = 300
local radius = 16
local vx = 150
local vy = 150

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Bouncing Ball")
end

function love.update(dt)
    x = x + vx * dt
    y = y + vy * dt

    if x - radius < 0 or x + radius > love.graphics.getWidth() then
        vx = -vx
    end

    if y - radius < 0 or y + radius > love.graphics.getHeight() then
        vy = -vy
    end
end

function love.draw()
    love.graphics.circle("fill", x, y, radius)
end